public/scripts/world/MapScene.js

```diff
@@ -0,0 +1,204 @@
+import {Scene} from '../Scene.js';
+import {Map} from './Map.js';
+import {Tooltip} from '../util/Tooltip.js';
+import {DIRECTION} from '../util/helper_functions.js';
+import {Game} from '../Game.js';
+
+export class MapScene extends Scene {
+  constructor(data) {
+    super(data);
+    this._layer = new Konva.Layer();
+
+    this.map = new Map({layer: this.layer, stage: data.stage});
+
+    this.layer.add(data.player.render);
+  }
+
+  get layer() {
+    return this._layer;
+  }
+
+  switchFrom() {
+    // This removes a child froma its parents without needing acces to the parent
+    this.layer.remove();
```

Comment:
> BUG: speed of players seems to double after switching from `FightScene` back to `MapScene`. Also shadows of the tooltip seems to be getting more and more dark? I think it could be rendering multiple?

@Fluffisaurus This is why - keybindings arent being removed after switching so they just keep adding in switchTo(). Good catch!

Reply:
Ahhhh ok. I think we should probably abstract it into a `Control` class and have a single instance of keybindings running have switchTo just call Control to `updateKeybindings` to a different set depending on the scene